### PR TITLE
allowed to reuse address

### PR DESCRIPTION
**Pull request by ShijunDeng**
allowed to reuse address

Fixes [#1920](https://github.com/reactor/reactor-netty/issues/1920)

### CODE REVIEW

File: resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java [hunk 1]
```diff
@@ -483,6 +483,7 @@ public DnsNameResolver(
         b.group(executor());
         b.channelFactory(channelFactory);
         b.option(ChannelOption.DATAGRAM_CHANNEL_ACTIVE_ON_REGISTRATION, true);
+        b.option(ChannelOption.SO_REUSEADDR, true);
```
Comment (chrisvest) [inline resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java:486]: I think we should only do this if the localAddress is not null and the port is non-zero.

**Review by chrisvest — COMMENTED** [review]
Please sign the ICLA

**Comment by violetagg** [comment]
@chrisvest @ShijunDeng What do you think about if there is a configuration for the options on the level of `DnsNameResolverBuilder`? Thus the one that configure `DnsNameResolver` will decide when to add this option.

**Comment by normanmaurer** [comment]
@violetagg yep... I wonder if we should go even further and allow to just pass in `ChannelOption`s.

**Comment by ShijunDeng** [comment]
> @chrisvest @ShijunDeng What do you think about if there is a configuration for the options on the level of `DnsNameResolverBuilder`? Thus the one that configure `DnsNameResolver` will decide when to add this option.

goog idea, I couldn't agree more

File: resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java (hunk 1 above)
Comment (ShijunDeng) [reply resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java]: done

**Comment by ShijunDeng** [comment]
> @ShijunDeng please let me know once you did the change and I will review again

In order not to confuse developers，how about doing the change in another PR？

**Comment by normanmaurer** [comment]
@ShijunDeng  I think there is no reason to pull this one in as it is if we do the `ChannelOption` change.

**Pull request closed without merging**